### PR TITLE
refactor: refactoring the initialization way of the MetaData

### PR DESCRIPTION
**Pull request by bigboss2063**
Using a chained call to initialize MetaData like NewMetaData.WithNewMetaData().WithKeySize(keySize).WithValueSize(valSize)...

### CODE REVIEW

File: inmemory/kv.go [hunk 1]
```diff
@@ -136,33 +136,16 @@ func put(shardDB *ShardDB, bucket string, key, value []byte, ttl uint32, flag ui
 	valueSize := uint32(len(value))
 	timestamp := uint64(time.Now().Unix())
 	bucketSize := uint32(len(bucket))
+	meta := nutsdb.NewMetaData().WithTimeStamp(timestamp).WithKeySize(keySize).WithValueSize(valueSize).WithFlag(flag).WithTTL(ttl).
+		WithBucketSize(bucketSize).WithStatus(nutsdb.Committed).WithDs(nutsdb.DataStructureBPTree)
 	err = shardDB.BPTreeIdx[bucket].Insert(key, &nutsdb.Entry{
```
Comment (elliotchenzichang) [inline inmemory/kv.go:141]: Seems this line can refactor to the new initial way of Entry, could you help to do me a favor?

File: inmemory/kv.go [hunk 2]
```diff
@@ -136,33 +136,16 @@ func put(shardDB *ShardDB, bucket string, key, value []byte, ttl uint32, flag ui
...
-			ValueSize:  valueSize,
-			Timestamp:  timestamp,
-			Flag:       flag,
-			TTL:        ttl,
-			BucketSize: bucketSize,
-			Status:     nutsdb.Committed,
-			Ds:         nutsdb.DataStructureBPTree,
-		},
+		Meta:   meta,
 	}, &nutsdb.Hint{
```
Comment (elliotchenzichang) [inline inmemory/kv.go:146]: Seems this line can refactor to the new initial way of the Hint object.  Would you like to do me a favor?

**Review by elliotchenzichang — COMMENTED** [review]
General LGTM. left some comments. Would you like to do me a favor? Thanks!

**Comment by bigboss2063** [comment]
I have completed.

**Comment by elliotchenzichang** [comment]
Merged, your coding capability is unbelievable.

**Pull request merged**